llmfoundry/utils/config_utils.py

```diff
@@ -178,9 +183,102 @@ def log_config(cfg: DictConfig) -> None:
+    """
+    data_paths = set()
+
+    for data_split in ['train', 'eval']:
+        data_set = cfg.get(f'{data_split}_loader', {}).get('dataset', {})
+        cfg_split = data_set.get('split', None)
+        source_dataset_path = cfg.get(f'source_dataset_{data_split}', {})
+
+        # Check for Delta table
+        if source_dataset_path and len(source_dataset_path.split('.')) >= 3:
+            data_paths.add(('delta_table', source_dataset_path, data_split))
+        # Check for UC volume
+        elif source_dataset_path and source_dataset_path.startswith('dbfs:'):
+            data_paths.add(('uc_volume', source_dataset_path[len('dbfs:'):], data_split))
+        # Check for HF path
+        elif 'hf_name' in data_set:
+            hf_path = data_set['hf_name']
+            backend, _, _ = parse_uri(hf_path)
+            if backend:
+                hf_path = f'{hf_path.rstrip("/")}/{cfg_split}' if cfg_split else hf_path
+                data_paths.add((backend, hf_path, data_split))
+            elif hf_path.startswith('/'):
+                data_paths.add(('local', hf_path, data_split))
+            else:
+                data_paths.add(('hf', hf_path, data_split))
+        # check for remote path
+        elif 'remote' in data_set:
+            remote_path = data_set['remote']
+            backend, _, _ = parse_uri(remote_path)
+            if backend:
+                remote_path = f'{remote_path.rstrip("/")}/{cfg_split}/' if cfg_split else remote_path
+                data_paths.add((backend, remote_path, data_split))
+            else:
+                data_paths.add(('local', remote_path, data_split))
+        else:
+            raise KeyError('DataSource Not Found')
+
+    return data_paths
+
+
+def _log_dataset_uri(cfg: DictConfig) -> None:
+    """Logs dataset tracking information to MLflow.
+
+    Args:
+        cfg (DictConfig): A config dictionary of a run
+    """
+    if mlflow is None:
+        raise ImportError('MLflow is not installed. Skipping dataset logging.')
+
+    # Figure out which data source to use
+    data_paths = _parse_source_dataset(cfg)
+
+    dataset_source_mapping = {
+        's3': mlflow.data.http_dataset_source.HTTPDatasetSource,
+        'oci': mlflow.data.http_dataset_source.HTTPDatasetSource,
+        'https': mlflow.data.http_dataset_source.HTTPDatasetSource,
+        'hf': mlflow.data.huggingface_dataset_source.HuggingFaceDatasetSource,
+        'delta_table': mlflow.data.delta_dataset_source.DeltaDatasetSource,
+        'uc_volume': mlflow.data.uc_volume_dataset_source.UCVolumeDatasetSource,
+        'local': mlflow.data.http_dataset_source.HTTPDatasetSource,
+    }
+
+    # Map data source types to their respective MLFlow DataSource.
+    for dataset_type, path, split in data_paths:
+
+        if dataset_type in dataset_source_mapping:
+            source_class = dataset_source_mapping[dataset_type]
+            if dataset_type == 'delta_table':
```

Comment:
instead of this if/else, can make the dictionary mapping above tuples, where the second item in the tuple is the path arg key name, and then just use that here